utilities/rate_limiters/write_amp_based_rate_limiter.cc

```diff
@@ -297,46 +307,39 @@ Status WriteAmpBasedRateLimiter::Tune() {
 
   int64_t prev_bytes_per_sec = GetBytesPerSecond();
 
-  bytes_sampler_.AddSample(duration_bytes_through_ * 1000 / duration_ms);
-  highpri_bytes_sampler_.AddSample(duration_highpri_bytes_through_ * 1000 /
-                                   duration_ms);
-  if (bytes_sampler_.AtTimePoint()) {
-    long_term_bytes_sampler_.AddSample(bytes_sampler_.GetFullValue());
-    long_term_highpri_bytes_sampler_.AddSample(
-        highpri_bytes_sampler_.GetFullValue());
+  for (uint32_t i = 0; i < duration_ms / kMillisPerTune; i++) {
+    bytes_sampler_.AddSample(duration_bytes_through_ * 1000 / duration_ms);
+    highpri_bytes_sampler_.AddSample(duration_highpri_bytes_through_ * 1000 /
+                                     duration_ms);
+    limit_bytes_sampler_.AddSample(prev_bytes_per_sec);
   }
-  limit_bytes_sampler_.AddSample(prev_bytes_per_sec);
-  // As LSM grows higher, it tends to generate compaction tasks in waves
-  // (cascaded). We use extra long-term window to help reduce this fluctuation.
   int32_t ratio = std::max(
-      kRatioLower, static_cast<int32_t>(
-                       long_term_bytes_sampler_.GetFullValue() * 10 /
-                       std::max(long_term_highpri_bytes_sampler_.GetFullValue(),
-                                kHighBytesLower)));
-  ratio = std::max(ratio, static_cast<int32_t>(
-                              bytes_sampler_.GetFullValue() * 10 /
-                              std::max(highpri_bytes_sampler_.GetFullValue(),
-                                       kHighBytesLower)));
-  int32_t ratio_padding = ratio * kRatioPaddingPercent / 100;
+      kRatioLower,
+      static_cast<int32_t>(
+          bytes_sampler_.GetFullValue() * 10 /
+          std::max(highpri_bytes_sampler_.GetFullValue(), kHighBytesLower)));
 
   // in case there are compaction bursts even when online writes are stable
-  auto util = bytes_sampler_.GetRecentValue() * 100 /
+  auto util = bytes_sampler_.GetRecentValue() * 1000 /
```

Comment:
```suggestion
  int64_t util = bytes_sampler_.GetRecentValue() * 1000 /
```
nit: avoid using `auto` for simple types for better readability.